frontend/src/pages/CompareV2.tsx

```diff
@@ -51,22 +69,102 @@ function CompareV2(props: PageProps) {
   const runIds = (queryParamRunIds && queryParamRunIds.split(',')) || [];
 
   // Retrieves run details.
-  const { isError, data, refetch } = useQuery<ApiRunDetail[], Error>(
+  const {
+    isLoading: isLoadingRunDetails,
+    isError: isErrorRunDetails,
+    error: errorRunDetails,
+    data: runs,
+    refetch,
+  } = useQuery<ApiRunDetail[], Error>(
     ['run_details', { ids: runIds }],
     () => Promise.all(runIds.map(async id => await Apis.runServiceApi.getRun(id))),
     {
       staleTime: Infinity,
-      onError: async error => {
-        const errorMessage = await errorToMessage(error);
+    },
+  );
+
+  // Retrieves MLMD states (executions and linked artifacts) from the MLMD store.
+  const {
+    data: runArtifacts,
+    isLoading: isLoadingRunArtifacts,
+    isError: isErrorRunArtifacts,
+    error: errorRunArtifacts,
+  } = useQuery<RunArtifacts[], Error>(
+    ['run_artifacts', { runIds, runs }],
+    () => {
+      if (runs) {
+        return Promise.all(
+          runIds.map(async (r, index) => {
+            const context = await getKfpV2RunContext(r);
+            const executions = await getExecutionsFromContext(context);
+            const artifacts = await getArtifactsFromContext(context);
+            const events = (await getEventsByExecutions(executions)).filter(
+              e => e.getType() === Event.Type.OUTPUT,
+            );
+
+            // Match artifacts to executions.
+            const artifactMap = new Map();
+            artifacts.forEach(artifact => artifactMap.set(artifact.getId(), artifact));
+            const executionArtifacts = executions.map(execution => {
+              const executionEvents = events.filter(e => e.getExecutionId() === execution.getId());
+              const linkedArtifacts = executionEvents.map(event => {
+                return {
+                  event,
+                  artifact: artifactMap.get(event.getArtifactId()),
+                } as LinkedArtifact;
+              });
+              return {
+                execution,
+                linkedArtifacts,
+              } as ExecutionArtifacts;
+            });
+            return {
+              run: runs[index],
+              executionArtifacts,
+            };
+          }),
+        );
+      }
+      return [];
+    },
+    {
+      staleTime: Infinity,
+    },
+  );
+
+  useEffect(() => {
+    if (isLoadingRunDetails || isLoadingRunArtifacts) {
+      return;
```

Comment:
Since the `useQuery` functions commonly re-run, meaning that the `isLoading` values toggle on and off relatively frequently within a number of seconds, I think this would make the banner update multiple times in just a few seconds. I think to simplify the page display, I'll keep the existing banner display until the `useQuery` function is no longer loading.

We still use the `info` mode in other cases (in `CompareV1.tsx` for example), but not as an intermediate state.

Reply:
Sounds good, it makes sense to me for not showing intermediate banner.